File: backend/src/main/java/cindyK/Eleven/toDoApp/backend/model/Task.java
package cindyK.Eleven.toDoApp.backend.model;

import javax.persistence.*;
import java.util.Date;

@Entity
@Table(name = "tasks")
public class Task {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    private String task;
    private String description;
    private Date published;
    private Date deadline;
    private int priority;
    private String status;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getTask() {
        return task;
    }

    public void setTask(String task) {
        this.task = task;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public Date getPublished() {
        return published;
    }

    public void setPublished(Date published) {
        this.published = published;
    }

    public Date getDeadline() {
        return deadline;
    }

    public void setDeadline(Date deadline) {
        this.deadline = deadline;
    }

    public int getPriority() {
        return priority;
    }

    public void setPriority(int priority) {
        this.priority = priority;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }
}

File: frontend/src/app/Task.ts
export class Task{
    id: number;
    task: string;
    deadline: Date;
    description: string;
    priority: number;
    status:string;

    constructor(    
        id: number,
        task: string,
        deadline: Date,
        description: string,
        priority: number,
        status:string
        ){
            this.id = id;
            this.task = task;
            this.deadline = deadline;
            this.description = description;
            this.priority = priority;
            this.status = status;
    }
}

